Model/Sync/CollectionsProducts/Processor.php

```diff
@@ -2,10 +2,352 @@
 
 namespace Yotpo\Core\Model\Sync\CollectionsProducts;
 
-class Processor
+use Exception;
+use Yotpo\Core\Model\AbstractJobs;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Store\Model\App\Emulation as AppEmulation;
+use Yotpo\Core\Model\Api\Sync as YotpoCoreSync;
+use Yotpo\Core\Model\Config as CoreConfig;
+use Yotpo\Core\Model\Sync\Catalog\Logger as CatalogLogger;
+use Yotpo\Core\Model\Sync\Catalog\YotpoResource;
+use Yotpo\Core\Model\Sync\CollectionsProducts\Services\CollectionsProductsService;
+use Yotpo\Core\Model\Sync\Category\Processor\Main as YotpoCategoryProcessorMain;
+use Yotpo\Core\Model\Sync\Category\Processor\ProcessByCategory as YotpoCategoryProcessor;
+use Yotpo\Core\Model\Sync\Catalog\Processor as YotpoProductProcessor;
+use Yotpo\Core\Model\Sync\Catalog\Processor\Main as YotpoProductProcessorMain;
+
+class Processor extends AbstractJobs
 {
+    const PRODUCT_SYNC_LIMIT_CONFIG_KEY = 'product_sync_limit';
+    const COLLECTIONS_PRODUCT_ENDPOINT_STRING = 'collections_product';
+    const YOTPO_COLLECTION_ID_REQUEST_PARAM_STRING = '{yotpo_collection_id}';
+    const LOGGER_LOG_ENTITY_FILE = 'catalog';
+
+    /**
+     * @var CoreConfig
+     */
+    protected $coreConfig;
+
+    /**
+     * @var YotpoCoreSync
+     */
+    protected $yotpoCoreSync;
+
+    /**
+     * @var CatalogLogger
+     */
+    protected $catalogLogger;
+
+    /**
+     * @var YotpoResource
+     */
+    protected $yotpoResource;
+
+    /**
+     * @var YotpoCategoryProcessorMain
+     */
+    protected $yotpoCategoryProcessorMain;
+
+    /**
+     * @var YotpoCategoryProcessor
+     */
+    protected $yotpoCategoryProcessor;
```

Comment:
`YotpoCategoryProcessor` extends `YotpoCategoryProcessorMain ` - no need for both.

Reply:
discussed offline, keeping it for readability and visibility.

Reply:
discussed offline, we're keeping it for readability and visibility